compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/transform.kt

```diff
@@ -36,6 +36,16 @@ fun <T : IrElement, D> MutableList<T>.transformInPlace(transformer: IrElementTra
     }
 }
 
+fun <T : IrElement, D> Array<T?>.transformInPlace(transformer: IrElementTransformer<D>, data: D) {
+    for (i in 0 until size) {
+        // Cast to IrElementBase to avoid casting to interface and invokeinterface, both of which are slow.
+        (get(i) as IrElementBase?)?.let {
+            @Suppress("UNCHECKED_CAST")
+            set(i, it.transform(transformer, data) as T)
+        }
+    }
```

Comment:
NB: `for (i in 0 until size)` is also in the function above.

Reply:
Pss waaaa no sé hacer esto

Reply:
`0 until size` -> `indices` is to fix the IDE inspection reported here because apparently it's an antipattern for iteration over arrays, but not lists (I don't know why). Both cases are generated as a counter loop by the JVM backend.